l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,249 @@
+    function newstateTransitionManager(address _stateTransitionManager) external onlyOwner {
+        require(!stateTransitionManagerIsRegistered[_stateTransitionManager], "Bridgehub: state transition already registered");
+        stateTransitionManagerIsRegistered[_stateTransitionManager] = true;
+    }
+
+    /// @notice State Transition can be any contract with the appropriate interface/functionality
+    /// @notice this stops new Chains from using the STF, old chains are not affected
+    function removestateTransitionManager(address _stateTransitionManager) external onlyOwner {
+        require(stateTransitionManagerIsRegistered[_stateTransitionManager], "Bridgehub: state transition already registered");
+        stateTransitionManagerIsRegistered[_stateTransitionManager] = false;
+    }
+
+    /// @notice token can be any contract with the appropriate interface/functionality
+    function newToken(address _token) external onlyOwner {
+        require(!tokenIsRegistered[_token], "Bridgehub: token already registered");
+        tokenIsRegistered[_token] = true;
+    }
+
+    /// @notice Bridge can be any contract with the appropriate interface/functionality
+    function newTokenBridge(address _tokenBridge) external onlyOwner {
+        require(!tokenBridgeIsRegistered[_tokenBridge], "Bridgehub: token bridge already registered");
+        tokenBridgeIsRegistered[_tokenBridge] = true;
+    }
+
+    function setWethBridge(address _wethBridge) external onlyOwner {
+        wethBridge = IL1Bridge(_wethBridge);
+    }
+
+    /// @notice for Eth the baseToken address is 0.
+    function newChain(
+        uint256 _chainId,
+        address _stateTransitionManager,
+        address _baseToken,
+        address _baseTokenBridge,
+        uint256 _salt,
+        address _l2Governor,
+        bytes calldata _initData
+    ) external onlyOwner returns (uint256 chainId) {
+        // KL TODO: clear up this formula for chainId generation
+        if (_chainId == 0) {
+            chainId = uint48(
+                uint256(
+                    keccak256(
+                        abi.encodePacked("CHAIN_ID", block.chainid, address(this), _stateTransitionManager, msg.sender, _salt)
+                    )
+                )
+            );
+        } else {
+            chainId = _chainId;
+        }
+
+        require(stateTransitionManagerIsRegistered[_stateTransitionManager], "Bridgehub: state transition not registered");
+        require(tokenIsRegistered[_baseToken], "Bridgehub: token not registered");
+        require(tokenBridgeIsRegistered[_baseTokenBridge], "Bridgehub: token bridge not registered");
+
+        require(stateTransitionManager[_chainId] == address(0), "Bridgehub: chainId already registered");
+
+        stateTransitionManager[chainId] = _stateTransitionManager;
+        baseToken[chainId] = _baseToken;
+        baseTokenBridge[chainId] = _baseTokenBridge;
```

Comment:
Difference is reading from calldata or storage.
```suggestion
        stateTransitionManager[_chainId] = _stateTransitionManager;
        baseToken[_chainId] = _baseToken;
        baseTokenBridge[_chainId] = _baseTokenBridge;
```

Reply:
No, the point is there is an input _chainId and an output chainId. Both are declared in the functions input/output arguements. Its not storage either way. Or am I wrong.